uwblueprint/rowan-house
# Use Apollo and gql for BaseAPI, Auth, Entity and Courses

## Implementation Description
Numerous API client fixes:
- Use `ApolloClient` in BaseAPIClient and leverage `ApolloProvider`
- Add AuthMutations file for Auth related gql strings
- Add EntitiesMutations and EntitiesQueries
- Remove LessonAPIClient and CourseAPIClient

## Screenshots
N/A

## What should reviewers focus on?

## Testing Procedure
- I tried all the Entities (create, update) and Auth (login, logout) functionality and it's working as expected

## Things to Note
- Additional dependencies/libraries you've added?
- Things you'd want to know when coming back to the code after a few months

## Checklist
